server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -256,6 +258,7 @@ public ThreadPool(
         );
         builders.put(Names.TRANSLOG_SYNC, new FixedExecutorBuilder(settings, Names.TRANSLOG_SYNC, allocatedProcessors * 4, 10000));
         builders.put(Names.REMOTE_PURGE, new ScalingExecutorBuilder(Names.REMOTE_PURGE, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)));
+        builders.put(Names.INDEX_SEARCHER, new FixedExecutorBuilder(settings, Names.INDEX_SEARCHER, allocatedProcessors, 1000, false));
```

Comment:
It should be done in `SearchModule` by the `ExecutorServiceProvider` instance in case the fall back to feature flag kicks in.

Reply:
Keeping the builder here but returning the provider in SearchModule which will return `INDEX_SEARCHER` threadpool in case when feature flag is set

Reply:
Please move it to SearchModule because it leaks outside even if the feature is turned off:

```
curl http://localhost:9200/_cat/thread_pool

...
runTask-0 index_searcher      0 0 0
runTask-0 listener            0 0 0
...

```

Reply:
Before `SearchModule` is created all the ExecutorBuilders are already constructed in `ThreadPool`, so don't see a way to add new ExecutorBuilder there. If I just return a `ExecutorService` in `SearchModule` then that won't be tracked by `ThreadPool` and also miss the stats around it. I considered adding Feature flag check in `ThreadPool` class before adding the builder in the map but I think we are trying to avoid that.

Reply:
Oh sorry, indeed the ExecutorBuilders were coming from PluginService, not a SearchPlugin,  adding feature flag check should make it, thanks.